pkg/api/openshiftcluster.go

```diff
@@ -803,4 +803,5 @@ type Identity struct {
 
 	Type                   string                 `json:"type,omitempty"`
 	UserAssignedIdentities UserAssignedIdentities `json:"userAssignedIdentities,omitempty"`
+	IdentityURL            string                 `json:"identityURL,omitempty"`
```

Comment:
Again this is the candidate for being `Mutable` here in InternalAPI only, as the field would be changed to an empty string after the bootstrap certificates are fetched.
Although that can be done when the story around the certificate fetching is performed later.
WDYT?

Reply:
Also will it fail here:- https://github.com/Azure/ARO-RP/pull/3514/files#diff-c0dceb05e2a59300f2e11380f3601ea682171f3f4837bb769689fc79e78b1a70R315
If the field is not `Mutable` for the internalAPI?

Reply:
I don't think so, since we're bailing out of the function early if the identity field isn't present: https://github.com/Azure/ARO-RP/pull/3514/files#diff-c0dceb05e2a59300f2e11380f3601ea682171f3f4837bb769689fc79e78b1a70R303 

Although, I could be wrong on that.

Reply:
I think I understand now - the concern is that we might not be able to modify IdentityURL once populated without that tag. I can add it since the assumption is that it's needed for fields that change (such as ServicePrincipalProfile, whose subfields use the mutable tag).

Reply:
Since we're talking about the internal API here, I don't think this tag is relevant; it can be present or absent and it will never be checked. The only place this tag gets checked is during static validation. See this function and its doc comment: https://github.com/Azure/ARO-RP/blob/master/pkg/api/util/immutable/immutable.go#L21-L24. The `mutable` tag is a thing we added to our codebase and not a Go thing.

I do see some fields with the `mutable` tag in the internal API though, so... keep it just to be safe?